src/BlockchainReader/Bitcoin.ts

```diff
@@ -44,6 +56,7 @@ const dataOutputToData = (dataOutput: VOutWithTxId): string => {
 
 export const dataToPoetAnchor = (data: string): PoetAnchor => {
   const buffer = Buffer.from(data, 'hex')
+  if (buffer.byteLength < anchorByteLength) return undefined
```

Comment:
don't really care for the idea of this function returning `undefined || PoetAnchor`.

I think some refactoring, then making the change is in order to simplify these functions, to allow us to add this check easier.

take a look at blockToPoetAnchors

```ts
export const blockToPoetAnchors = (block: Block): ReadonlyArray<PoetBlockAnchor> =>
  block.tx
    .map(transactionToPoetAnchor)
    .filter(isTruthy)
    .filter(poetAnchorHasCorrectPrefix)
    .map(poetAnchorWithBlockData(block))
```

We can refactor and fix some of the functions it uses to be simpler by making them responsible for a single thing

Something like this

```ts
export const blockToPoetAnchors = (block: Block): ReadonlyArray<PoetBlockAnchor> =>
  block.tx
    .map(transactionToDataOutput)
    .filter(isTruthy)
    .map(dataOutputToData)
    .map(dataToBuffer)
    .map(bufferToPoetAnchor)
    .filter(poetAnchorHasCorrectPrefix)
    .map(poetAnchorWithBlockData(block))
```

That allows us to then add a filter function that checks the buffer length.

```ts
export const blockToPoetAnchors = (block: Block): ReadonlyArray<PoetBlockAnchor> =>
  block.tx
    .map(transactionToDataOutput)
    .filter(isTruthy)
    .map(dataOutputToData)
    .map(dataToBuffer)
    .filter(isCorrectBufferLength) // we can now simply add this new filter
    .map(bufferToPoetAnchor)
    .filter(poetAnchorHasCorrectPrefix)
    .map(poetAnchorWithBlockData(block))
```

Refactoring in a separate PR first, then make the change.

Reply:
Ok. Would you be OK with approving this PR and I will open one with the refactor? This is blocking the QA environment testing I believe.

Reply:
I am going to say no unless other disagree. I would like to do the refactor first, then the fix. We have a bad habit of rushing changes to code because its blocking something else, in reality, something is almost always blocking something else, we need to start tackling issues better at some point and I think we should start now so we start forming those better habits.